### PR TITLE
Show # commits ahead or behind remote branch

### PR DESCRIPTION
**Pull request by arielcrivas**
Show number of commits ahead or behind remote branch alongside arrow

### CODE REVIEW

File: pure.zsh [hunk 1]
```diff
@@ -78,8 +78,23 @@ prompt_pure_check_git_arrows() {
 	arrow_status=(${(ps:\t:)arrow_status})
 	local arrows left=${arrow_status[1]} right=${arrow_status[2]}
 
-	(( ${right:-0} > 0 )) && arrows+="${PURE_GIT_DOWN_ARROW:-⇣}"
-	(( ${left:-0} > 0 )) && arrows+="${PURE_GIT_UP_ARROW:-⇡}"
+	if (( ${right:-0} > 0 )); then
+        commits=$(command git rev-list --count HEAD..@{upstream} 2>/dev/null)
```
Comment (mafredri) [inline pure.zsh]: This information is already available in `$right`, no need for a re-executing `git rev-list`.

File: pure.zsh [hunk 2]
```diff
@@ -78,8 +78,23 @@ prompt_pure_check_git_arrows() {
...
+    fi
+
+	if (( ${left:-0} > 0 )); then
+        commits=$(command git --count HEAD..@{upstream} 2>/dev/null)
+        if [[ $commits ]] && (( commits )); then
+            arrows+="${commits}${PURE_GIT_UP_ARROW:-⇡}"
+        else
+            arrows+="${PURE_GIT_UP_ARROW:-⇡}"
+        fi
+    fi
```
Comment (mafredri) [inline pure.zsh]: We use tabs for indentation.

File: pure.zsh [hunk 3]
```diff
@@ -78,8 +78,23 @@ prompt_pure_check_git_arrows() {
...
+        if [[ $commits ]] && (( commits )); then
+            arrows+="${commits}${PURE_GIT_DOWN_ARROW:-⇣}"
+        else
+            arrows+="${PURE_GIT_DOWN_ARROW:-⇣}"
+        fi
+    fi
+
+	if (( ${left:-0} > 0 )); then
+        commits=$(command git --count HEAD..@{upstream} 2>/dev/null)
+        if [[ $commits ]] && (( commits )); then
```
Comment (mafredri) [inline pure.zsh]: `[[ ]]` is used to check for quality, e.g. `[[ $commits -gt 0 ]]`, I'm not even sure what this checks for. Here also just a `(( commits ))` would suffice, but the `(( left > 0 ))` already check the same thing.

**Comment by sindresorhus** [comment]
Could be added to the wiki.

**Pull request closed without merging**